src/storage/key_prefix.rs

```diff
@@ -9,24 +9,65 @@ pub const TIDB_RANGES_COMPLEMENT: &[(&[u8], &[u8])] =
 pub const RAW_KEY_PREFIX: u8 = b'r';
 pub const TXN_KEY_PREFIX: u8 = b'x';
 
-/// Checks if the key is in TiDB encode.
+/// KeyPrefixToCheck
 ///
-/// Returning true doesn't mean that the key is certainly written by
-/// TiDB, but instead, it matches the definition of TiDB key in API V2,
-/// therefore, the key is treated as TiDB data in order to fulfill the
-/// compatibility.
-pub fn is_tidb_key(key: &[u8]) -> bool {
-    matches!(KeyPrefix::parse(key).0, KeyPrefix::TiDB)
+/// Accept different kinds of keys, to avoid tedious casting at caller side.
+pub trait KeyPrefixToCheck {
```

Comment:
We may implement this trait for any type can deref as &[u8]

Reply:
Could you give more detail ?

I had tried to using coercion here but failed. `KeyPrefixToCheck` is invoked in [`check_api_version()`](https://github.com/pingyu/tikv/blob/98d7c823b97a83cbc1f823f5b3cef3966b190b27/src/storage/mod.rs#L377) and callers pass keys by trait `IntoIterator<Item = xx>`, where the "deref" is not performed.

From https://doc.rust-lang.org/nomicon/coercions.html:
> Note that we do not perform coercions when matching traits...

Reply:
I think this is not idiomatic to rust. The caller is responsible for taking care of the ownership, and this kind of automatics increases the difficulty for the reader about which implementation on earth is used.

Reply:
PTAL, thanks~ @iosmanthus @andylokandy